lib/ansible/modules/cloud/docker/docker_host_facts.py

```diff
@@ -0,0 +1,293 @@
+    from docker.errors import APIError, NotFound
+except ImportError:
+    # missing docker-py handled in ansible.module_utils.docker_common
+    pass
+
+try:
+    from ansible.module_utils.docker_common import docker_version, clean_dict_booleans_for_docker_api
+except Exception as dummy:
+    # missing docker-py handled in ansible.module_utils.docker
+    pass
+
+
+class DockerHostManager(DockerBaseClass):
+
+    def __init__(self, client, results):
+
+        super(DockerHostManager, self).__init__()
+
+        self.client = client
+        self.results = results
+
+        listed_objects = ['volumes', 'networks', 'containers', 'images']
+
+        self.results['docker_host_facts'] = self.get_docker_host_facts()
+
+        if self.client.module.params['disk_usage'] is True:
+            self.results['docker_disk_usage'] = self.get_docker_disk_usage_facts()
+
+        for docker_object in listed_objects:
+            if self.client.module.params[docker_object] is True:
+                returned_name = "docker_" + docker_object + "_list"
+                filter_name = docker_object + "_filters"
+                filters = clean_dict_booleans_for_docker_api(client.module.params.get(filter_name))
+                self.results[returned_name] = self.get_docker_items_list(docker_object, filters)
+
+    def get_docker_host_facts(self):
+        try:
+            return self.client.info()
+        except APIError as exc:
+            self.client.fail_json(msg="Error inspecting docker host: %s" % to_native(exc))
+
+    def get_docker_disk_usage_facts(self):
+        try:
+            return self.client.df()['LayersSize']
+        except APIError as exc:
+            self.client.fail_json(msg="Error inspecting docker host: %s" % to_native(exc))
+
+    def get_docker_items_list(self, docker_object=None, filters=None):
+        items = None
+        items_list = []
+
+        header_containers = ['Id', 'Image', 'Command', 'Created', 'Status', 'Ports', 'Names']
+        header_volumes = ['Driver', 'Name']
+        header_images = ['Id', 'RepoTags', 'Created', 'Size']
+        header_networks = ['Id', 'Driver', 'Name', 'Scope']
+
+        try:
+            if docker_object == 'containers':
+                items = self.client.containers(filters=filters)
+            if docker_object == 'networks':
+                items = self.client.networks(filters=filters)
+            if docker_object == 'images':
+                items = self.client.images(filters=filters)
+            if docker_object == 'volumes':
+                items = self.client.volumes(filters=filters)
+        except APIError as exc:
+            self.client.fail_json(msg="Error inspecting docker host: %s" % to_native(exc))
+
+        if docker_object != 'volumes':
+            for item in items:
+                item_record = dict()
+
+                if docker_object == 'containers':
+                    for key in header_containers:
+                        item_record[key] = item.get(key)
```

Comment:
Why just return a subset and not everything?

Reply:
It is a decision on the idea of this module. It may return all the whole information or match the `ls` command output. I picked up the second way because if you perform advanced tasks you have to call an appropriate `_facts` module anyway. This leaves output here having the essentials and small if you need to print it or something. But I don't mind changing it and return full structure obtained this way if that would be the preference for maintainers of other `docker_*` modules.

Reply:
It's a good question. On one hand, it's nice to have a clean list which isn't littered with extra 1000 fields you're not interested in. On the other hand, it's very annoying if you need something which is returned by docker but thrown away by the module, and you need to loop over the result and feed it into `docker_*_facts` just to get the value back.

How about adding a `verbose_list_elements: yes|no` option which toggles between returning a subset and everything? Then one can enable full output if one needs it.

Reply:
Sounds good